src/ucs/datastruct/ptr_array.c

```diff
@@ -54,16 +49,37 @@ ucs_ptr_array_freelist_set_next(ucs_ptr_array_elem_t *elem, unsigned next)
                     (((ucs_ptr_array_elem_t)next) << UCS_PTR_ARRAY_NEXT_SHIFT);
 }
 
+static inline unsigned
+ucs_ptr_array_freelist_get_size_free_ahead(ucs_ptr_array_elem_t elem)
+{
+    ucs_assert(__ucs_ptr_array_is_free(elem));
+    return (elem >> UCS_PTR_ARRAY_SIZE_FREE_SHIFT);
+}
+
+static inline void
+ucs_ptr_array_freelist_element_set_free(ucs_ptr_array_elem_t *elem,
```

Comment:
plz rename this func, since it will be the only one containing `freelist` in the name.
maybe `ucs_ptr_array_size_free_set`? (similar to `ucs_ptr_array_size_free_get`)

Reply:
Renamed to: ucs_ptr_array_freelist_element_set()